src/main/java/org/isf/opd/gui/OpdBrowser.java

```diff
@@ -421,43 +437,125 @@ private JButton getJDeleteButton() {
+	
+	private JPanel getButtonsPanel() {
+		JPanel buttonsPanel = new JPanel();
+		JPanel filterButtonPanel = new JPanel();
+		filterButtonPanel.add(getFilterButton());
+		filterButtonPanel.add(getResetButton());
+		buttonsPanel.setLayout(new BoxLayout(buttonsPanel, BoxLayout.Y_AXIS));
+		buttonsPanel.add(filterButtonPanel);
+		buttonsPanel.add(getRowCounter());
+		return buttonsPanel;
+	}
 
+	private JPanel getOtherFiltersPanel() {
+		JPanel panel = new JPanel(new SpringLayout());
+		panel.setBorder(BorderFactory.createTitledBorder(MessageBundle.getMessage("angal.opd.otherfilters.border")));
+		panel.add(getWardBoxPanel());
+		panel.add(getJSelectionDiseasePanel());
+		panel.add(getSexPanel());
+		panel.add(getDateFilterPanel());
+		panel.add(getJAgePanel());
+		panel.add(getPatientTypePanel());
+		if (isSingleUser) {
+			panel.add(Box.createVerticalGlue());
+			SpringUtilities.makeCompactGrid(panel, 7, 1, 5, 5, 5, 5);
+		} else {
+			panel.add(getUserPanel());
+			panel.add(Box.createVerticalGlue());
+			SpringUtilities.makeCompactGrid(panel, 8, 1, 5, 5, 5, 5);
+		}
+		return panel;
+	}
+	
+	private JPanel getWardBoxPanel() {
+		JPanel wardBoxPanel = new JPanel();
+		wardBoxPanel.setLayout(new BoxLayout(wardBoxPanel, BoxLayout.Y_AXIS));
+		wardBoxPanel.add(getWardBox());
+		return wardBoxPanel;
+	}
+
+	public JPanel getUserPanel() {
+		JPanel userPanel = new JPanel();
+		groupUserFilter = new ButtonGroup();
+		radioMyPatients = new JRadioButton(MessageBundle.getMessage("angal.opd.mypatient.btn"));
+		radioAllPatients = new JRadioButton(MessageBundle.getMessage("angal.common.all.btn"));
+		radioAllPatients.setSelected(true);
+		groupUserFilter.add(radioMyPatients);
+		groupUserFilter.add(radioAllPatients);
+		userPanel.add(radioMyPatients);
+		userPanel.add(radioAllPatients);
+		return userPanel;
+	}
+	
+	private Component getSearchCodesPanel() {
+		JPanel searchCodesPanel = new JPanel(new SpringLayout());
+		searchCodesPanel.setBorder(BorderFactory.createTitledBorder(MessageBundle.getMessage("angal.opd.searchbycodespressenter.border")));
+		opdCodeFilter = new JTextField(3);
+		opdCodeFilter.addKeyListener(new SearchByOPDCodeListener());
+		progYearFilter = new JTextField(3);
+		progYearFilter.addKeyListener(new SearchByProgYearListener());
+		patientCodeFilter = new JTextField(3);
+		patientCodeFilter.addKeyListener(new SearchByPatientIdListener());
+		searchCodesPanel.add(new JLabel(MessageBundle.getMessage("angal.common.code.txt")));
+		searchCodesPanel.add(new JLabel(MessageBundle.getMessage("angal.opd.opdnumber.txt")));
+		searchCodesPanel.add(new JLabel(MessageBundle.getMessage("angal.common.patientID")));
+		searchCodesPanel.add(opdCodeFilter);
+		searchCodesPanel.add(progYearFilter);
+		searchCodesPanel.add(patientCodeFilter);
+		SpringUtilities.makeCompactGrid(searchCodesPanel, 2, 3, 5, 5, 5, 5);
+		return searchCodesPanel;
+	}
+
+	private JButton getResetButton() {
+		if (resetButton == null) {
+			resetButton = new JButton("Reset");
```

Comment:
shouldn't this be a property string and not a hard code literal?

Reply:
Good catch! Let me fix that! Thanks!